teps/0011-redirecting-step-output-streams.md

```diff
@@ -0,0 +1,219 @@
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Option 1: Allowing Users to Specify Redirection Paths](#option-1-allowing-users-to-specify-redirection-paths)
+    - [Example Usage](#example-usage)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Option 2: Redirecting to Canonical Conventional Paths](#option-2-redirecting-to-canonical-conventional-paths)
+    - [Example Usage](#example-usage-1)
+    - [Risks and Mitigations](#risks-and-mitigations-1)
+- [Design Details](#design-details)
+- [Alternatives](#alternatives)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Consuming outputs of a step in another step is a common pattern in writing Tasks. However, this is currently tedious to do. Task authors have to overwrite the image entrypoint with either `sh -c` or the `script` field to wrap up the command to run with an explicit output stream redirection. This is not even possible if the image does not come with a shell, e.g., images built by [`ko`](https://github.com/google/ko).
+
+To achieve the functionality of output stream redirection between steps and even Tasks, Tekton will add new fields to steps for Task authors to specify paths to redirect stdout/stderr to.
+
+## Motivation
+
+Allow Task authors to support the following use cases:
+
+* Run image `gcr.io/k8s-staging-boskos/boskosctl` with args in a step and process its output through another `jq` image in a subsequent step to acquire the project name without overwriting the image entrypoint, building custom image with both utilities, or looking into container logs ([tektoncd/pipeline#2925](https://github.com/tektoncd/pipeline/issues/2925)).
+
+* Run image `alpine/git` with args `["describe", "--tags"]` in a step and extract the stdout as a Task result without overwriting the image entrypoint to be used by subsequent tasks.
+
+* In general, enable Task authors to access stdout/stderr of a step easily in another step in the same or different Task, without requiring authors to: 1) overwrite image entrypoints; 2) write wrapping scripts solely to do output stream redirection; and 3) build wrapper images to include all needed utilities.
+
+By enabling redirection between steps and Task results, it is much easier for Task authors to composite utility images in a Task.
+
+### Goals
+
+1. Allow the stdout/stderr of a step to be consumed by another step.
+1. Enable a user to configure the path where the output streams are written.
+
+### Non-Goals
+
+1. Parse stdout/stderr of a step into a structured format (e.g., JSON) and extract information from certain fields.
+
+## Requirements
+
+* Add new fields for a step to specify paths to redirect stdout/stderr to.
+* Users should be able to observe the output streams through Pod logs even if `StdoutPath` or `StderrPath` is specified. In other words, output streams should be duplicated instead of simply redirected.
```

Comment:
Having thought about this a bit more - I wonder if "hiding" or limiting the stdout log would be more a runtime concern. Something to configure at the taskrun level instead of task.  I'm o.k. to move ahead with the existing proposal of just the path string!